source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -103,12 +103,13 @@ void Filter::setDecoderFilterCallbacks(Http::StreamDecoderFilterCallbacks& callb
   decoding_state_.setDecoderFilterCallbacks(callbacks);
   const Envoy::StreamInfo::FilterStateSharedPtr& filter_state =
       callbacks.streamInfo().filterState();
-  if (!filter_state->hasData<ExtProcLoggingInfo>(ExtProcLoggingInfoName)) {
-    filter_state->setData(ExtProcLoggingInfoName, std::make_shared<ExtProcLoggingInfo>(),
+  if (!filter_state->hasData<ExtProcLoggingInfo>(callbacks.filterConfigName())) {
+    filter_state->setData(callbacks.filterConfigName(),
+                          std::make_shared<ExtProcLoggingInfo>(config_->metadata()),
                           Envoy::StreamInfo::FilterState::StateType::Mutable,
                           Envoy::StreamInfo::FilterState::LifeSpan::Request);
   }
-  logging_info_ = filter_state->getDataMutable<ExtProcLoggingInfo>(ExtProcLoggingInfoName);
```

Comment:
I think this is a maybe candidate for runtime guard. The argument for it is that we can't actually detect the breakage at build time and this might just magically break someone. However, since this filter is still technically in alpha state https://github.com/envoyproxy/envoy/blob/122661e80f69d0165173020af1286a36b66df292/source/extensions/extensions_metadata.yaml#L319, I don't feel we're strongly committed to production stability yet. 

It might make sense to update version history and leave it at that.

Reply:
This is reason I was suggesting bool flag to maintain backwards compatibility. I don't feel strongly on this though

Probably we could update the ext_proc_filter rst with this change.